specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2024-04-01-preview/keyvault.json

```diff
@@ -0,0 +1,2411 @@
+      "enum": [
+        "Pending",
+        "Approved",
+        "Rejected",
+        "Disconnected"
+      ],
+      "x-ms-enum": {
+        "name": "PrivateEndpointServiceConnectionStatus",
+        "modelAsString": true
+      }
+    },
+    "PrivateEndpointConnectionProvisioningState": {
+      "type": "string",
+      "readOnly": true,
+      "description": "The current provisioning state.",
+      "enum": [
+        "Succeeded",
+        "Creating",
+        "Updating",
+        "Deleting",
+        "Failed",
+        "Disconnected"
+      ],
+      "x-ms-enum": {
+        "name": "PrivateEndpointConnectionProvisioningState",
+        "modelAsString": true
+      }
+    },
+    "PrivateLinkResourceListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "Array of private link resources",
+          "items": {
+            "$ref": "#/definitions/PrivateLinkResource"
+          }
+        }
+      },
+      "description": "A list of private link resources",
+      "type": "object"
+    },
+    "PrivateLinkResource": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PrivateLinkResourceProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "A private link resource",
+      "type": "object"
+    },
+    "PrivateLinkResourceProperties": {
+      "properties": {
+        "groupId": {
+          "description": "Group identifier of private link resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "requiredMembers": {
+          "description": "Required member names of private link resource.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "requiredZoneNames": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Required DNS zone names of the the private link resource."
+        }
+      },
+      "description": "Properties of a private link resource.",
+      "type": "object"
+    },
+    "NetworkSecurityPerimeterAssociationsConfigurationListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "Array of network security perimeter associations",
+          "items": {
+            "$ref": "#/definitions/NetworkSecurityPerimeterAssociationsConfigurationResult"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of associations."
+        }
+      },
+      "description": "A of network security perimeter associations",
+      "type": "object"
+    },
+    "NetworkSecurityPerimeterAssociationsConfigurationResult": {
```

Comment:
>NetworkSecurityPerimeterAssociationsConfigurationResult [](http://example.com/codeflow?start=5&length=55)

For NSP, the hope is that, we can refactor to $ref of these definitions soon

[https://github.com/Azure/azure-rest-api-specs/pull/28958](https://github.com/Azure/azure-rest-api-specs/pull/28958)